src/sendgrid_java_wrapper/core.clj

```diff
@@ -15,17 +15,41 @@
          email))
 
 (defn send-email
-  [auth {to :to from :from subject :subject html :html}]
+  [auth {to :to from :from subject :subject html :html from-name :from-name}]
   (let [email (-> (prepare-email from subject html)
-                 (.addTo to))
+                  (.addTo to)
+                  (.setFromName from-name))
+        response (-send auth email)]
+    (.getMessage response)))
+
+(defn send-email-groupid
+  [auth {to :to from :from subject :subject html :html from-name :from-name group-id :group-id}]
+  (let [email (-> (prepare-email from subject html)
+                  (.addTo to)
+                  (.setFromName from-name)
+                  (.setASMGroupId group-id))
         response (-send auth email)]
     (.getMessage response)))
 
 (defn bulk-email
-  [auth {bcc :bcc from :from subject :subject html :html}]
+  [auth {bcc :bcc from :from  subject :subject html :html from-name :from-name}]
+  (println "[bulk email bcc] " bcc )
+  (let [email (-> (prepare-email from subject html)
+                  (.setFromName from-name)
+                  (.setBcc (into-array String bcc)))
+        hack (dorun
+              (println "the super awesome email" (bean email))
+              (flush))
+        response (-send auth email)]
+    (.getMessage response)))
+
+(defn bulk-email-groupid
+  [auth {bcc :bcc from :from  subject :subject html :html from-name :from-name group-id :group-id}]
   (println "[bulk email bcc] " bcc )
   (let [email (-> (prepare-email from subject html)
-                 (.setBcc (into-array String bcc)))
+                  (.setFromName from-name)
```

Comment:
from-name will have the name the sender want their receivers to have instead of just the email. So, they will say Webtalk instead of no_reply@webtalk.co. I already tested, if you send null nothing happens.